mlrun/feature_store/feature_set.py

```diff
@@ -1030,6 +1030,61 @@ def deploy_ingestion_service(
             self, source, targets, name, run_config, verbose
         )
 
+    def extract_relation_keys(
+        self,
+        other_feature_set,
+        relation: Dict[str, Union[str, Entity]] = None,
+    ) -> list[str]:
+        """
+        This method checks if the other feature set can be merged to the right of this feature set
+
+        :param other_feature_set:   The feature set you want to merge to the right of this feature set
+        :param relation:            All the relation that where defined on this feature set.
+
+        :return:                    If those 2 feature set are merge able it returns a list of the left join keys
+                                    otherwise it returns an empty list.
+                                    (The right join keys are always the entities of the other feature set)
+        """
+        right_feature_set_entity_list = other_feature_set.spec.entities
+
+        if all(
+            ent in self.spec.entities for ent in right_feature_set_entity_list
+        ) and len(right_feature_set_entity_list) == len(self.spec.entities):
+            # entities wise
+            return list(self.spec.entities.keys())
+        elif all(ent in self.spec.entities for ent in right_feature_set_entity_list):
+            # entities wise when the right fset have lower number of entities
+            return list(right_feature_set_entity_list.keys())
+
+        elif relation:
```

Comment:
Looks like this case is not tested.

Reply:
@gtopper test named ` test_relation_join` is testing this case.